jaggery-modules/store/scripts/user.js

```diff
@@ -7,18 +7,18 @@ var user = {};
 
     var USER_ROLE_PREFIX = 'Internal/private_';
 
-    var cleanUsername = function (username) {
+    user.cleanUsername = function (username) {
         /**
          * this is a one-way hash function, @ is replaced if the user name is an email
          * if the user name is coming from a secondery user store it will be second.com/user hence
          * "/" will be replaced.
          */
 
-        return username.replace('@', ':').replace('/', ':');
+        return username.replace('@', '-AT-').replace('/', ':');
```

Comment:
We test this js (with -AT-) with identity 5.0.9-SNAPSHOT, kernal-4.4.5 and registry -4.5.3 versions without any issue.